### ISSUE TITLE
Support multiline configuration for `jsx-equals-spacing`

### ISSUE THREAD

**Issue by iandunn** [yannickcr/eslint-plugin-react#2030]
I think it's desirable to have different `jsx-equals-spacing` config for single and multiple lines, similar to [`key-spacing`](https://eslint.org/docs/rules/key-spacing). 

For example, I'd like this to treated as correct:

```javascript
<PanelBody title={ __( 'Which Speakers?', 'wordcamporg' ) } initialOpen={ true }>
	<PanelRow>
		<CheckboxControl
			label    = { __( 'Show all', 'wordcamporg' ) }
			checked  = { show_all_posts }
			onChange = { ( value ) => setAttributes( { show_all_posts: value } ) }
		/>
	</PanelRow>
</PanelBody>
```

When the element's attributes are on a single line, there should be no spaces around the `=`, but when they are on multiple lines, there should be at `1` space after the `=`, and `1+` spaces before the `=` so that the `=` are aligned.

Currently, I don't see any way to achieve that; it's either all or nothing.

**Comment by ljharb** [comment]
This style of alignment, while common in other languages, is wildly unidiomatic and uncommon for JS. Can you point to any popular style guides where this is common?

Separately, if the multi line example is going to force at least 1 space around the =, then I’d expect the single-line example to do the same. Alignment, obviously, would only apply when multiline.

**Comment by iandunn** [comment]
Alright, sounds like I'll need to write a plugin. Thanks.
